analyzer/src/main/kotlin/managers/Npm.kt

```diff
@@ -108,7 +111,9 @@ open class Npm(
 
         val npmRcFile = Os.userHomeDirectory.resolve(".npmrc")
         if (npmRcFile.isFile) {
-            ortProxySelector.addProxies(managerName, readProxySettingsFromNpmRc(npmRcFile.readText()))
+            val npmRcContent = npmRcFile.readText()
+            ortProxySelector.addProxies(managerName, readProxySettingsFromNpmRc(npmRcContent))
+            npmRegistry = readNpmRegistryFromNpmRc(npmRcContent)
```

Comment:
If `readNpmRegistryFromNpmRc()` / `readRegistryFromNpmRc()` would return a nullable string, we could rewrite the above code as:

    private val npmRegistry: String

    // ...

    init {
        val npmRcFile = Os.userHomeDirectory.resolve(".npmrc")
        npmRegistry = if (npmRcFile.isFile) {
            val npmRcContent = npmRcFile.readText()
            ortProxySelector.addProxies(managerName, readProxySettingsFromNpmRc(npmRcContent))
            readNpmRegistryFromNpmRc(npmRcContent) ?: PUBLIC_NPM_REGISTRY
        } else {
            PUBLIC_NPM_REGISTRY
        }
    }

which would also get rid of the `var` in favor of a `val`.